variants/devise-2fa/template.rb

```diff
@@ -0,0 +1,50 @@
+# Allow us to copy file with root at the directory this file is in
+source_paths.unshift(File.dirname(__FILE__))
+
+def print_header(msg)
+  puts "=" * 80
+  puts msg
+  puts "=" * 80
+end
+
+print_header "Adding devise-two-factor, rqrcode-rails3 to Gemfile"
+run "bundle add devise-two-factor"
+run "bundle add rqrcode-rails3"
+
+print_header "Adding OTP info to user model"
+run "bundle exec rails generate devise_two_factor User DEVISE_TWO_FACTOR_SECRET_ENCRYPTION_KEY"
+
+gsub_file("app/models/user.rb", "ENV['DEVISE_TWO_FACTOR_SECRET_ENCRYPTION_KEY']", "Rails.application.secrets.devise_two_factor_secret_encryption_key")
+
+insert_into_file("config/secrets.yml", after: /\A.+secret_key_base.+\z/) do
+  <<-EO_LINE
+    devise_two_factor_secret_encryption_key: "<%= ENV['DEVISE_TWO_FACTOR_SECRET_ENCRYPTION_KEY'] %>"
+  EO_LINE
+end
+
+append_to_file ".env" do
```

Comment:
should this also write to example.env?

Reply:
Good catch - it should. This PR has changed a lot and will change more before it's ready for review again but I have made a note of this.